### PR TITLE
Dfldb

### PR DESCRIPTION
**Pull request by RayMan504**
setup helpers for database User and Friend request database queries. setup express routers for users and friend requests

### CODE REVIEW

**Review by chusto2 — APPROVED** [review]

File: dark-matter-disco/database-postgres/helpers.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  return User.findOrCreate({ 
+    where: { username }, 
+    defaults: {
+      username,
+      starsTotal: 0,
+      status: 0,
+    }     
+  }).then((success) => {
+    console.log(success[0].id, 'stored');
+    return success[0].id
```
Comment (jesseLessinger) [inline dark-matter-disco/database-postgres/helpers.js]: why return only the id and not the whole user?

File: dark-matter-disco/database-postgres/helpers.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  })
+};
+
+
+
+//get user by name
+
+const getUserByUsername = (username) => {
+  return User.findAll({ where: { username } })
+    .then(data => data);
```
Comment (jesseLessinger) [inline dark-matter-disco/database-postgres/helpers.js]: does this .then do anything?

File: dark-matter-disco/database-postgres/helpers.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+const getUserById = (id) => {
+  return User.findAll({ where: { id } }).then(data => data);
```
Comment (jesseLessinger) [inline dark-matter-disco/database-postgres/helpers.js]: does findAll return an array? why not just find or findOne which ever it is in sequelize

File: dark-matter-disco/database-postgres/helpers.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+const getPendingRequests = (userId) => {
+  return Friends.findAll({ where: { userId, status: 0 }}).then(pendingRequests => pendingRequests);
```
Comment (jesseLessinger) [inline dark-matter-disco/database-postgres/helpers.js]: i think you have to match the userId that coming in as a param, with the friendId in the friends table, a pending request is a request made by someone else to me

File: dark-matter-disco/database-postgres/helpers.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    return success
+  });
+}
+
+
+
+const acceptFriendRequest = (userId, friendId) => {
+  //update status on accepted friend request
+  Friends.update({status: 1}, { where: { userId, friendId }  })
+}
```
Comment (jesseLessinger) [inline dark-matter-disco/database-postgres/helpers.js:58]: this looks good, but do we also need to add the second entry with the friend relationship going the other way

File: dark-matter-disco/server/routes/friends.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+const express = require('express');
+const router = express.Router();
+const { storeFriendRequest, getPendingRequests, acceptFriendRequest, } = require('../../database-postgres/helpers.js');
+
+//setup post request for friend invitations
+router
+  .get('/', (req, res) => {
+    //call helper to find user by username
+    getPendingRequests(1).then((results) => {
```
Comment (jesseLessinger) [inline dark-matter-disco/server/routes/friends.js]: this can't be hardcoded

File: dark-matter-disco/server/routes/friends.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        //send user as response
+        res.send(results);
+    });
+  })
+  .post('/', (req, res) => {
+    //call helper to save pending requests
+    storeFriendRequest(1, 2);
```
Comment (jesseLessinger) [inline dark-matter-disco/server/routes/friends.js]: make dynamic too

File: dark-matter-disco/server/routes/friends.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    storeFriendRequest(2, 1);
+
+    //send 201 status code
+    res.send(201);
+  })
+//setup post for accepted friends
+.put('/', (req, res) => {
+  // storeFriendRequest(2, 1).then(() => {
+      acceptFriendRequest(1, 2);
```
Comment (jesseLessinger) [inline dark-matter-disco/server/routes/friends.js]: dynamic

File: dark-matter-disco/server/routes/users.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+const express = require('express');
+const router = express.Router();
+const { storeOrFindUser, getUserByUsername, getUserById, } = require('../../database-postgres/helpers.js');
+
+
+router.get('/', (req, res) => {
+  //call helper to find user by username
+  getUserById(2).then((result) => {
```
Comment (jesseLessinger) [inline dark-matter-disco/server/routes/users.js]: dynamic

File: dark-matter-disco/src/app/dance-floor/dance-floor.component.ts [hunk 10]
```diff
@@ -93,13 +93,17 @@ export class DanceFloorComponent implements AfterViewInit, OnInit {
 
     // send user pose data to friends
     this.userPoseStream.subscribe((poses) => {
+      socketService.emit('pose', poses, 'James');
```
Comment (jesseLessinger) [inline dark-matter-disco/src/app/dance-floor/dance-floor.component.ts]: can't mess with code like this any push it up

**Review by jesseLessinger — CHANGES_REQUESTED** [review]

File: dark-matter-disco/database-postgres/helpers.js (hunk 2 above)
Comment (RayMan504) [reply dark-matter-disco/database-postgres/helpers.js]: yes. I need to return a single user once the client initiates a get request

File: dark-matter-disco/database-postgres/helpers.js (hunk 5 above)
Comment (RayMan504) [reply dark-matter-disco/database-postgres/helpers.js:58]: I added the second call so that so that both friend relationships are updated in the database

**Review by jesseLessinger — APPROVED** [review]

**Pull request merged**